# This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
# If a copy of the MPL was not distributed with this file, You can obtain one at
# https://mozilla.org/MPL/2.0/. Adapts jank-lang/clojure-test-suite for the Suss harness:
# exact binary64 :suss branches for the JavaScript Number range constants.
--- a/test/clojure/core_test/number_range.cljc
+++ b/test/clojure/core_test/number_range.cljc
@@ -5,6 +5,7 @@
 (def ^:const max-int #?(:cljr Int64/MaxValue
                         :phel php/PHP_INT_MAX
                         :jank (#cpp (:member (std.numeric_limits jank.i64) max))
+                        :suss 9007199254740991
                         :cljs js/Number.MAX_SAFE_INTEGER
                         :clj Long/MAX_VALUE
                         :default 0x7FFFFFFFFFFFFFFF))
@@ -12,6 +13,7 @@
 (def ^:const min-int #?(:cljr Int64/MinValue
                         :phel php/PHP_INT_MIN
                         :jank (#cpp (:member (std.numeric_limits jank.i64) min))
+                        :suss -9007199254740991
                         :cljs js/Number.MIN_SAFE_INTEGER
                         :clj Long/MIN_VALUE
                         :default -0x8000000000000000))
@@ -29,6 +31,7 @@
                            :lpy (.-max sys/float-info)
                            :phel php/PHP_FLOAT_MAX
                            :jank (#cpp (:member (std.numeric_limits jank.f64) max))
+                           :suss 1.7976931348623157e308
                            :cljs js/Number.MAX_VALUE
                            :clj Double/MAX_VALUE
                            :default 1.7976931348623157e+308))
@@ -37,6 +40,7 @@
                            :lpy (.-min sys/float_info)
                            :phel php/PHP_FLOAT_MIN
                            :jank (#cpp (:member (std.numeric_limits jank.f64) min))
+                           :suss 5e-324
                            :cljs js/Number.MIN_VALUE
                            :clj Double/MIN_VALUE
                            :default 4.9e-324))
